### PR TITLE
Fix README.txt for release packages

### PR DESCRIPTION
**Pull request by Wengier**
I have noticed that all current packages including the release zip packages contain file README/README.TXT which claims they are development snapshot builds with some technical info. This PR adds a new README/README.TXT file as a brief introduction to the project for inclusion with the release packages (with tags like refs/tags/v0.79.0) instead of the file for development snapshot builds.

### CODE REVIEW

**Comment by kcgen** [comment]
Thanks @Wengier, the end result is good.

The `README` is already a template file consistent of common text plus placeholder variables populated from metadata at package-creation time:

**docs/README.template**
`<snip>`
`Please report any bugs or issues to: https://github.com/`**`%GITHUB_REPO%`**
`...`
`branch:`**`%GIT_BRANCH%`**
`commit:`**`%GIT_COMMIT%`**
`...`
`Have fun!`

(We've even parameterized the REPO, so this file is properly created in forks).

The PR's approach of adding a new release-only README and flip-flopping:



doesn't make use of the template, which can be expanded with the improved wording in your new README, plus new field(s) to indicate if it's a release tag (or a development build).

Ideally we want the template fed from as much repo metadata as possible.

**Comment by kcgen** [comment]
Regarding determining if the currently checkout is a release or not, I think that logic should be moved into the package creation script or a stand-alone script.  

The PR does this using three duplicate and not-obvious shell lines in the CI code:



If a user run the script locally, they won't have those three shell lines, because they only exist in the CI files. 

The `git describe --exact-match HEAD` command will succeed and return the  current commit's annotated tag value (such as `v0.78.1`, if we're on a release) or the command will fail otherwise (if we're on an untagged dev commit). 

Something like this can describe the type of commit:

``` shell
if version_tag=$(git describe --exact-match HEAD 2> /dev/null); then
    echo release $version_tag
else
   echo a development branch
fi
```

could be put into the package-creation script, and its result could be injected into the improved README template.

`This package contains`**`%COMMIT_DESCRIPTION%`**

**Comment by kcgen** [comment]
Regarding determining if the commit is a release or not:

> `export opt=
          if [[ "${{ github.ref }}" == "refs/tags/"* ]] && [[ "${{ github.ref }}" != *"-"* ]]; then export opt=-t; fi`


The `git describe --exact-match HEAD` command will succeed and return the  current commit's annotated tag value (such as `v0.78.1`, if we're on a release) or the command will fail otherwise (if we're on an untagged dev commit). 

Something like this can describe the type of commit:

``` shell
if version_tag=$(git describe --exact-match HEAD 2> /dev/null); then
    echo release $version_tag
else
   echo a development branch
fi
```

could be put into the package-creation script, and its result could be injected into the improved README template.

`This package contains`**`%COMMIT_DESCRIPTION%`**

**Comment by kcgen** [comment]
Regarding the `-t` flag, I believe it's not needed once the package creation script can describe the current commit and populate the template with that result.

File: docs/README.template [hunk 1]
```diff
@@ -1,10 +1,21 @@
-This is the development snapshot build of dosbox-staging project.
+This package contains %PACKAGE_INFORMATION% of the DOSBox Staging project.
```
Comment (kcgen) [inline docs/README.template:1]: Looks great!

File: scripts/create-package.sh [hunk 2]
```diff
@@ -4,20 +4,20 @@ set -e
...
         -h          : Show this help message
-        -p          : Buld platform. Can be one of linux, macos, msys2, msvc
+        -p          : Build platform. Can be one of linux, macos, msys2, msvc
         -c          : Git commit
         -b          : Git branch
         -r          : Git repository
-        -v          : dosbox-staging version
-        -f          : force creation if PACKAGE_DIR is not empty
+        -v          : DOSBox Staging version
+        -f          : Force creation if PACKAGE_DIR is not empty
```
Comment (kcgen) [inline scripts/create-package.sh:20]: nice cleanups :+1:

File: scripts/create-package.sh [hunk 3]
```diff
@@ -74,11 +74,16 @@ install_doc()
...
+        export packge_information="release $version_tag"
+    elif [ -n "$git_branch" ] && [ -n "$git_commit" ]; then
+        export packge_information="a development branch named $git_branch with commit $git_commit"
+    else
+        export packge_information="a development branch"
     fi
-    if [ -n "$git_branch" ]; then
-        sed -i -e "s|%GIT_BRANCH%|$git_branch|" "$readme_tmpl"
+    if [ -n "$packge_information" ]; then
+        sed -i -e "s|%PACKAGE_INFORMATION%|$packge_information|" "$readme_tmpl"
```
Comment (kcgen) [inline scripts/create-package.sh:86]: (small typo in `package_information`)

Comment (Wengier) [reply scripts/create-package.sh]: Fixed typo. Thanks for pointing out.

File: scripts/create-package.sh [hunk 4]
```diff
@@ -74,11 +74,16 @@ install_doc()
...
     # Fill template variables in README.template
-    if [ -n "$git_commit" ]; then
-        sed -i -e "s|%GIT_COMMIT%|$git_commit|" "$readme_tmpl"
+    if [[ "$git_branch" == "refs/tags/"* ]] && [[ "$git_branch" != *"-"* ]]; then
+        export version_tag=`echo $git_branch | awk '{print substr($0,11);exit}'`
+        export package_information="release $version_tag"
+    elif [ -n "$git_branch" ] && [ -n "$git_commit" ]; then
+        export package_information="a development branch named $git_branch with commit $git_commit"
+    else
+        export package_information="a development branch"
```
Comment (kcgen) [inline scripts/create-package.sh:83]: The `export` keyword will place the `package_information` variable in the calling shell's environment (is this an intended side-effect of the script?).

If it's not intended, then we can drop the `export` and use local variables.

(Sorry for the delay on this comment @Wengier , I had it queued up yesterday but didn't submit it).

**Review by kcgen — COMMENTED** [review]
Thanks @Wengier.
Just one comment regarding use of `export`.

File: scripts/create-package.sh (hunk 4 above)
Comment (Wengier) [reply scripts/create-package.sh]: Thanks for pointing this out. I removed the export keyword, which hopefully makes the variables local.

**Review by kcgen — APPROVED** [review]
Looks good!

**Pull request merged**